crates/router/src/core/payments/operations/payment_create.rs

```diff
@@ -763,6 +770,35 @@ impl PaymentCreate {
                 .await
             })
             .await;
+
+        if additional_pm_data.is_none() {
+            // If recurring payment is made using payment_method_id, then fetch payment_method_data from retrieved payment_method object
+            additional_pm_data = payment_method_info
+                .as_ref()
+                .async_map(|pm_info| async {
+                    domain::types::decrypt::<serde_json::Value, masking::WithType>(
+                        pm_info.payment_method_data.clone(),
+                        key_store.key.get_inner().peek(),
+                    )
+                    .await
+                    .change_context(errors::StorageError::DecryptionError)
+                    .attach_printable("unable to decrypt card details")
+                    .ok()
```

Comment:
why are we ignoring the error here?

Reply:
If the decryption of payment_method_data, we don't want to fail the payment